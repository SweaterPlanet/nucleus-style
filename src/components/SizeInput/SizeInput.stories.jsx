import React from 'react'
import { withKnobs, boolean, number, text } from '@storybook/addon-knobs/react'
import SizeInput from './index'

export default {
  title: 'SizeInput',
  component: SizeInput,
  decorators: [withKnobs],
}

export const Default = () => (
  <SizeInput
    dark={boolean('dark', false)}
    label={text('label', 'xs')}
    placeholder={number('placeholder', '0')}
    red={boolean('red', false)}
    stock={number('stock', '')}
  />
)

export const Dark = () => (
  <SizeInput
    dark={boolean('dark', true)}
    label={text('label', 'xs')}
    placeholder={number('placeholder', '0')}
    red={boolean('red', false)}
    stock={number('stock', '')}
  />
)

export const Group = () => (
  <>
    <SizeInput
      dark={boolean('dark', false)}
      label="xs"
      placeholder={number('placeholder', '0')}
      stock={number('stock', '')}
    />
    <SizeInput
      dark={boolean('dark', false)}
      label="sm"
      placeholder={number('placeholder', '0')}
      stock={number('stock', '')}
    />
    <SizeInput
      dark={boolean('dark', false)}
      label="md"
      placeholder={number('placeholder', '0')}
      stock={number('stock', '')}
    />
    <SizeInput
      dark={boolean('dark', false)}
      label="lg"
      placeholder={number('placeholder', '0')}
      stock={number('stock', '')}
    />
    <SizeInput
      dark={boolean('dark', false)}
      label="xl"
      placeholder={number('placeholder', '0')}
      stock={number('stock', '')}
    />
  </>
)

export const Caption = () => (
  <SizeInput
    dark={boolean('dark', false)}
    label={text('label', 'xs')}
    placeholder={number('placeholder', '0')}
    red={boolean('red', true)}
    stock={number('stock', '3')}
  />
)
