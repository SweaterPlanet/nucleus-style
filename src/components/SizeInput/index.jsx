import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import theme from '../../theme'

import Flex from '../Flex'

const Box = styled(Flex)`
  width: 72px;
  height: 36px;
  background-color: ${theme.colors.paper.dark};
  margin: 0 auto;
  position: absolute;
  direction: row;
`

const Label = styled.p`
  font-family: ${theme.fonts.body};
  font-size: 10px;
  font-weight: bold;
  padding: 3px;
  padding-left: 12px;
  color: ${theme.colors.slate.base};
  text-align: left;
`

const Input = styled.input`
  font-family: ${theme.fonts.body};
  font-size: 16px;
  font-weight: bold;
  border: 0;
  outline: 0;
  background: transparent;
  text-align: center;
  border-bottom: 1px solid ${theme.colors.slate.base};
  width: 20px;
  height: 16px;
  margin: 8px;
  margin-left: 8px;

  /* Hide number input errors */
  -moz-appearance: textfield;

  ::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  ::-webkit-outer-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  /* Placeholder styles */
  ::placeholder {
    color: ${theme.colors.slate.base};
    font-size: 16px;
    padding: 0;
    margin: 0;
  }

  ::-webkit-input-placeholder {
    color: ${theme.colors.slate.base};
    font-size: 16px;
    padding: 0;
    margin: 0;
  }

  :-ms-input-placeholder {
    color: ${theme.colors.slate.base};
    font-size: 16px;
    padding: 0;
    margin: 0;
  }
`

function SizeInput(props) {
  return (
    <Box>
      <Label bold>XS</Label>
      <Input type="number" placeholder="0" />
    </Box>
  )
}

export default SizeInput
