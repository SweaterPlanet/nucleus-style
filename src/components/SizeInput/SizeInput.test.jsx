import React from 'react'
import { mount, shallow } from 'enzyme'

import SizeInput from '.'

describe('SizeInput', () => {
  it('renders correctly', () => {
    const test = shallow(<SizeInput />)
    expect(test).toMatchSnapshot()
  })

  it('respects the dark prop', () => {
    const testProp = true
    const test = mount(<SizeInput dark={testProp} />)
    expect(test.prop('dark')).toBe(true)
  })

  it('respects the disabled prop', () => {
    const testProp = true
    const test = mount(<SizeInput disabled={testProp} />)
    expect(test.prop('disabled')).toBe(true)
  })

  it('respects the label prop', () => {
    const testProp = 'xs'
    const test = mount(<SizeInput label={testProp} />)
    expect(test.prop('label')).toBe(testProp)
  })

  it('respects the placeholder prop', () => {
    const testProp = 3
    const test = mount(<SizeInput placeholder={testProp} />)
    expect(test.prop('placeholder')).toBe(testProp)
  })

  it('respects the red prop', () => {
    const testProp = true
    const test = mount(<SizeInput red={testProp} />)
    expect(test.prop('red')).toBe(true)
  })

  it('respects the stock prop', () => {
    const testProp = 4
    const test = mount(<SizeInput stock={testProp} />)
    expect(test.prop('stock')).toBe(testProp)
  })
})
