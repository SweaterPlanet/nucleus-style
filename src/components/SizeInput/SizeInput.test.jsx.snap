// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SizeInput renders correctly 1`] = `ShallowWrapper {}`;
